### PR TITLE
Add a dashed option, with thickness and changes for modifying obstructed. Also added code paths. Tested.

### PR DESCRIPTION
**Pull request by gponick**
Add a dashed option, with thickness and code paths.

Also added obstructed options.

Tested locally.

### CODE REVIEW

File: BTMLColorLOSMod/WeaponRangeIndicators_DrawLine_Patch.cs [hunk 1]
```diff
@@ -189,16 +189,24 @@ public static class WeaponRangeIndicators_DrawLine_Patch
...
-                            line.material.color = Color.white;
-                            Color color6 = Color.Lerp(
-                                Color.clear,
-                                BTMLColorLOSMod.ModSettings.IndirectLineOfFireArcColor, 
-                                shotQuality);
-                            line.endColor = (line.startColor = color6);
+                            if (BTMLColorLOSMod.ModSettings.IndirectLineOfFireArcDashed)
+                            {
+                                line.material = __instance.MaterialOutOfRange;
+                                line.material.color = BTMLColorLOSMod.ModSettings.IndirectLineOfFireArcColor;
```
Comment (janxious) [inline BTMLColorLOSMod/WeaponRangeIndicators_DrawLine_Patch.cs]: We probably don't want to lose the shotQuality bit, so maybe something more like pseudocode:
```
if dashed
  set new materials
else
  just set white material color
do the normal lerp'ing
```
If the line color setting messes with the display of the dashed line color, maybe that is not the thing to do. THoughts?

Comment (gponick) [reply BTMLColorLOSMod/WeaponRangeIndicators_DrawLine_Patch.cs]: 5m :) I wasn't sure how you felt about that so I didn't want to touch it. Refactoring.

File: BTMLColorLOSMod/WeaponRangeIndicators_DrawLine_Patch.cs [hunk 2]
```diff
@@ -189,16 +201,25 @@ public static class WeaponRangeIndicators_DrawLine_Patch
...
-                            line.material.color = Color.white;
                             Color color6 = Color.Lerp(
                                 Color.clear,
-                                BTMLColorLOSMod.ModSettings.IndirectLineOfFireArcColor, 
+                                BTMLColorLOSMod.ModSettings.IndirectLineOfFireArcColor,
                                 shotQuality);
-                            line.endColor = (line.startColor = color6);
+                            if (BTMLColorLOSMod.ModSettings.IndirectLineOfFireArcDashed)
+                            {
+                                line.material = __instance.MaterialOutOfRange;
```
Comment (janxious) [inline BTMLColorLOSMod/WeaponRangeIndicators_DrawLine_Patch.cs:215]: So the default line width is too small to make this readable. I popped in:
```
line.endWidth = __instance.LOSWidthEnd * 1.75f;      // dashes are harder to see,
line.startWidth = __instance.LOSWidthBegin * 1.75f;  // so make 'em bigger
```

Original width:
screen shot 2018-05-21 at 12 28 51 am

Bigger width:
screen shot 2018-05-21 at 12 40 23 am

File: BTMLColorLOSMod/Settings.cs [hunk 3]
```diff
@@ -10,8 +10,43 @@ public float[] indirectLineOfFireArcColor
             set => IndirectLineOfFireArcColor = new Color(value[0], value[1], value[2], value[3]);
         }
 
+        public float[] blockedLOFLine1Color
```
Comment (janxious) [inline BTMLColorLOSMod/Settings.cs]: I'd like to see the setters grouped with their backing variable, and comments to explain what ideas all these represent.

File: BTMLColorLOSMod/WeaponRangeIndicators_DrawLine_Patch.cs [hunk 4]
```diff
@@ -162,15 +163,26 @@ public static class WeaponRangeIndicators_DrawLine_Patch
                                 Vector3 collisionPoint = previewInfo.collisionPoint;
                                 collisionPoint = Vector3.Project(collisionPoint - vector, vector2 - vector) + vector;
                                 line.SetPosition(1, collisionPoint);
+                                if(BTMLColorLOSMod.ModSettings.BlockedLOFLine1Active)
```
Comment (janxious) [inline BTMLColorLOSMod/WeaponRangeIndicators_DrawLine_Patch.cs]: do we need this setting?

Comment (gponick) [reply BTMLColorLOSMod/WeaponRangeIndicators_DrawLine_Patch.cs]: I figured some would want the first half of the line red, the second half a different color; and some would want both lines different colors. This just preserves the original code path if you don't want the first half changed.

I don't feel strongly.

File: BTMLColorLOSMod/Settings.cs (hunk 3 above)
Comment (gponick) [reply BTMLColorLOSMod/Settings.cs]: Updated this but maybe we should wrap the whole section of each "type" in a region? Or nest them in the mod.json? Up to you.

**Pull request merged**